feat: complete HL perp trading implementation

Minnzen opened the pull request:
<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * Multi-mode Close Position modal (Market & Limit) with mid-price assist; Set TP/SL modal; Withdraw flow from Perp accounts; per-order cancel and cancel-all.

* **Improvements**
  * Consolidated trading inputs and forms (Price, Size, TP/SL, TradingFormInput); stronger validation/formatting; trading button disabled for zero/invalid size; updated side-based styling and icon colors.

* **Bug Fixes**
  * Decimal defaulting preserved (0 retained where intended).
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

Minnzen committed:
refactor: restructure TradingPanel components with modular organization

- Reorganize TradingPanel components into logical subdirectories (inputs, modals, panels, selectors)
- Add comprehensive index.ts for clean API exports and better developer experience
- Improve import paths and reduce coupling between components
- Enhance form state management with automatic reset on token switching
- Consolidate component structure for better maintainability (net -1354 lines)
- Implement unified component export pattern across trading interface
- Optimize ClosePositionModal with enhanced functionality and validation
- Add reduceOnly order support to trading operations

Minnzen committed:
refactor: centralize trading component styling with unified utility functions

- Refactor PerpTradingPanel and TradeSideToggle to use shared style functions
- Enhance OrderConfirmModal with consistent color theming and button styles
- Improve ClosePositionModal decimal formatting with proper precision
- Unify ITradeSide type usage across trading components
- Reduce code duplication and improve style consistency across trading interface

Kahnchan committed:
refactor: update TradingPanel components for improved consistency and accessibility (#8327)

- Change button sizes and border radii for uniformity across TradingPanel components
- Update icon fill color to use 'currentColor' for better theming support
- Refactor text sizes for improved readability and accessibility
- Remove unnecessary props and streamline component structure for maintainability
- Enhance layout spacing and alignment for a more cohesive user interface

Minnzen committed:
refactor: extract TpslInput component and simplify trading form

- Add dedicated TpslInput for TP/SL price and percentage management
- Remove 200+ lines of calculation logic from PerpTradingForm
- Consolidate price validation and formatting functions in tokenUtils
- Improve code organization with BigNumber-based calculations

Minnzen committed:
feat: implement comprehensive TP/SL system with provider architecture

- Add setPositionTpsl API in ServiceHyperliquidExchange for advanced order management
- Implement SetTpslModal for user-friendly TP/SL configuration on existing positions
- Add PerpsProvider system with dedicated context store for improved state isolation
- Integrate TP/SL functionality into PerpPositionsList with action handlers
- Add IPositionTpslOrderParams type for structured TP/SL order parameters
- Enhance TpslInput component with better calculation accuracy and UX
- Simplify PerpTradingPanel architecture by removing redundant logic

Minnzen committed:
refactor: improve account management and UI polish

- Unify account handling with usePerpUseChainAccount across components
- Fix toast message grammar for proper singular/plural forms
- Add padding to Slider component in LeverageAdjustModal for better UX
- Simplify useHyperliquidTrading hook by removing redundant state management
- Clean up imports and unused dependencies

Minnzen committed:
feat: enhance deposit/withdraw functionality and improve trading validation

- Add comprehensive withdrawal support with dedicated withdraw action and IWithdrawParams type
- Centralize Hyperliquid constants (deposit address, USDC info, min amounts) to perp.constants
- Integrate DepositWithdrawModal with PerpsProvider system for better state management
- Improve trading button validation with size > 0 requirement and better disabled states
- Fix UI text consistency: "Not Enough Margin" and adjust button size to medium
- Optimize resetTradingForm to preserve user preferences while clearing transaction data
- Add _placeOrder method with proper builder integration for order processing

Minnzen committed:
Merge branch 'x' into perp-trading

Minnzen committed:
Merge branch 'x' into perp-trading

Minnzen committed:
refactor: migrate PerpTradersHistoryListModal to PerpsProviderMirror

- Replace ProviderJotaiContextHyperliquid with PerpsProviderMirror for consistency
- Use EJotaiContextStoreNames.perps for standardized store naming

Minnzen committed:
refactor: consolidate utility functions and simplify service layer

- Move Perp utility functions from local utils to shared perpsUtils (254→122 lines)
- Simplify ServiceHyperliquidExchange by removing redundant code and improving structure
- Unify type naming: IMarketOrderCloseParams → IOrderCloseParams for consistency
- Update all component imports to use centralized @onekeyhq/shared/src/utils/perpsUtils
- Add proper error handling with getter-based exchangeClient access pattern
- Remove duplicate validation logic and helper text generation from components
- Enhance code reusability by centralizing price formatting and validation functions

Minnzen committed:
Merge branch 'x' into perp-trading

Minnzen committed:
refactor: improve number formatting in position modals

- Enhance formatWithPrecision to remove trailing zeros and unnecessary decimal points
- Update ClosePositionModal and SetTpslModal to use unified formatting function

Minnzen committed:
refactor: remove tokenUtils export from Perp utils index

- Clean up unused export after tokenUtils migration to shared package

revan-zhang commented:
### :tada: **Snyk checks have passed. No issues have been found so far.**

:white_check_mark: **security/snyk** check is complete. No issues have been found. [(View Details)](https://app.snyk.io/org/rayston92/pr-checks/004b6c1f-2a65-4a9c-a031-772e8f122ab3?refs=106032d04fb9ea0dc20e52653cb4d0aa2166d5d7&source=prChecksComment)

:white_check_mark: **license/snyk** check is complete. No issues have been found. [(View Details)](https://app.snyk.io/org/rayston92/pr-checks/004b6c1f-2a65-4a9c-a031-772e8f122ab3/license?refs=106032d04fb9ea0dc20e52653cb4d0aa2166d5d7&source=prChecksComment)

originalix committed:
Merge branch 'x' into perp-trading

originalix committed:
Merge branch 'x' into perp-trading

Minnzen committed:
feat: add token persistence and improve leverage modal

- Add currentToken persistence to SimpleDbEntityPerp
- Implement memoized currentTokenAtom with storage sync
- Enhance LeverageAdjustModal UI and functionality
- Update formatWithPrecision with optional trailing zero removal

Minnzen committed:
Merge branch 'perp-trading' of https://github.com/OneKeyHQ/app-monorepo into perp-trading

Minnzen committed:
Merge branch 'x' into perp-trading

originalix commented on packages/shared/src/utils/perpsUtils.ts:
```diff
@@ -285,13 +285,135 @@ function calculateSpreadPercentage(
+}
+
 export {
+  MAX_DECIMALS_PERP,
```
之后的 PR 挪到 perp.constants 里

Minnzen replied:
ok

originalix commented on packages/shared/src/utils/perpsUtils.ts:
```diff
@@ -285,13 +285,135 @@ function calculateSpreadPercentage(
 function formatWithPrecision(
   value: string | number | BigNumber,
   decimals: number,
+  removeTrailingZeros = false,
```
函数用对象传参，这样在阅读调用方的代码时比较直观

也可以后续的 PR 再优化

Minnzen replied:
ok ， 3个参数想着不多，那之后达到3个都用对象传参

originalix replied:
不用按参数数量作为指标，统一走对象参数的设计是可以的，其他工具类也是这么设计的，比如 `accountUtils`

Minnzen replied:
ok
